File: Export/Natives/Name/Factories/IFileNamePartProviderContainer.cs
using System;
using FileNaming.CustomFileNaming;

namespace kCura.WinEDDS.Core.Export.Natives.Name.Factories
{
	public interface IFileNamePartProviderContainer
	{
		IFileNamePartProvider GetProvider(DescriptorPart descriptor);

		void Register(Type descriptorPartType, IFileNamePartProvider provider);
	}
}

File: FileNaming/CustomFileNaming/IFileNamePartProviderContainer.vb
Imports FileNaming.CustomFileNaming

Namespace kCura.WinEDDS.FileNaming.CustomFileNaming
	Public Interface IFileNamePartProviderContainer

		Function GetProvider(descriptor As DescriptorPart) As IFileNamePartProvider

		Sub Register(descriptorPartType As Type, provider As IFileNamePartProvider)

	End Interface
End Namespace